test/exe/Main.hs

```diff
@@ -977,6 +978,30 @@ pluginTests = testSessionWait "plugins" $ do
       )
     ]
 
+cppTests :: TestTree
+cppTests = testSessionWait "cpp" $ do
+  let content =
+        T.unlines
+          [ "{-# LANGUAGE CPP #-}",
+            "module Testing where",
+            "#ifdef FOO",
+            "foo = 42"
+          ]
+  _ <- openDoc' "Testing.hs" "haskell" content
+  expectDiagnostics
+    [ ( "Testing.hs",
+        [ ( DsError,
+            (2, -1),
+            T.unlines
+              [ " error: unterminated #ifdef",
+                " #ifdef FOO",
+                " ^"
```

Comment:
My CPP doesn’t seem to have the carret in the output here so it would probably be good to make this test a bit more fuzzy and only assert on `error: unterminated #ifdef` which is hopefully present in the output of most if not all CPP implementations.

Reply:
Reduced to `error: unterminated` because the error message on my mac is `error: unterminated conditional directive`.